### ISSUE TITLE
[MNG-6605] Allow to suppress download messages in interactive mode

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8608]
**[Gunnar Wagenknecht](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gunnar)** opened **[MNG-6605](https://issues.apache.org/jira/browse/MNG-6605?redirect=false)** and commented

When running Maven in batch mode (with option `-B`) it's possible to suppress download messages using "`-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn`"

Example:
`mvn clean install -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn`

When leaving out the `-B` option this no longer works.

**Example:**

```
export MAVEN_OPTS='-Xmx1g -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Dorg.slf4j.simpleLogger.showDateTime=true -Dorg.slf4j.simpleLogger.dateTimeFormat=HH:mm:ss'
mvn clean install
```

Prints out time stamps as configured but does not suppress download progress messages.

**Expected:**
Suppressing "Downloaded" messages from the console output should be possible without requiring the "`--batch-mode`" command line argument.

---

**Affects:** 3.6.0

**Attachments:**
- [2019-03-06_14-06-09.png](https://issues.apache.org/jira/secure/attachment/12961393/2019-03-06_14-06-09.png) (_174.52 kB_)

**Issue Links:**
- [MRELEASE-965](https://issues.apache.org/jira/browse/MRELEASE-965) Upload completes but doesn't update stdout - making it look like a freeze

**Remote Links:**
- [GitHub Pull Request #239
](https://github.com/apache/maven/pull/239)

3 votes, 6 watchers

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

`Slf4jMavenTransferListener` is not used in interactive mode. It writes to [`stdout`](https://github.com/apache/maven/blob/master/maven-embedder/src/main/java/org/apache/maven/cli/transfer/ConsoleMavenTransferListener.java).

It is not clear what you really request.

**Comment by jira-importer** [comment]
**[Gunnar Wagenknecht](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gunnar)** commented

From looking at the output it seems that `ConsoleMavenTransferListener` is used in addition to `Slf4jMavenTransferListener` or some other listener that prints "Downloading/Downloaded" messages to stdout. At least, that's how Maven output looks like in the Travis CI build logs. Please have a look the attached screenshot.

!2019-03-06_14-06-09.png!

I run the Maven command without `--batch-mode` and I do get those "Downloading/Downloaded" messages. However, I'd like a way to suppress them without having to specify `--batch-mode`.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

You cannot suppress them unless you go in batch mode. if you want to retain colors, you have to force them because stdout is not attached to a tty.

**Comment by jira-importer** [comment]
**[Raymond Augé](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rotty3000)** commented

I've submitted a potential solution here https://github.com/apache/maven/pull/239

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I still don't understand your problem and will not merge the PR. You want batch mode with colors. I recommended to force colors, did you try that?

**Comment by jira-importer** [comment]
**[Raymond Augé](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rotty3000)** commented

[Gunnar Wagenknecht](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gunnar) could yon confirm if my proposal solves your requirement for this issue?

[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o) IIUC this has nothing to do with colors. It is purely for silencing download messages (such as these):

```java
Downloading from central: https://repo.maven.apache.org/maven2/biz/aQute/bnd/biz.aQute.bnd.maven/4.2.0/biz.aQute.bnd.maven-4.2.0.pom
Downloaded from central: https://repo.maven.apache.org/maven2/biz/aQute/bnd/biz.aQute.bnd.maven/4.2.0/biz.aQute.bnd.maven-4.2.0.pom (3.6 kB at 1.7 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/biz/aQute/bnd/biz.aQute.bndlib/4.2.0/biz.aQute.bndlib-4.2.0.pom
Downloaded from central: https://repo.maven.apache.org/maven2/biz/aQute/bnd/biz.aQute.bndlib/4.2.0/biz.aQute.bndlib-4.2.0.pom (2.1 kB at 4.6 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/biz/aQute/bnd/biz.aQute.bnd.maven/4.2.0/biz.aQute.bnd.maven-4.2.0.jar
Downloaded from central: https://repo.maven.apache.org/maven2/biz/aQute/bnd/biz.aQute.bnd.maven/4.2.0/biz.aQute.bnd.maven-4.2.0.jar (112 kB at 61 kB/s)
```

during day to day use of maven (i.e. during "interactive" mode).

Truth be told the only time this information is of any real value is when an error occur which my change does neither prevent or improve the current state. Also the progress of downloading is also still provided. Furthermore, this is an opt in mode which requires setting `-Dorg.apache.maven.cli.transfer.ConsoleMavenTransferListener.quiet`.

I hope this explains my take on this issue. :)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I understand it now and reject the PR. I would expect someone to introduce `--no-transfer-progress`/`--disable-transfer-progress` and say in `MavenCli`:

```java
if ( quiet || disableTransferProgress )
{
    transferListener = new QuietMavenTransferListener();
}
```

**Comment by jira-importer** [comment]
**[Raymond Augé](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rotty3000)** commented

Ok great! At least with this feedback I can change the PR to do what is suggested! :)

thank you!

**Comment by jira-importer** [comment]
**[Raymond Augé](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rotty3000)** commented

[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o) I've updated the PR to reflect your suggested approach.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [95401cf7a606daa0982c3fe5a5928cd466230995](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=95401cf7a606daa0982c3fe5a5928cd466230995).
